src/solver.rs

```diff
@@ -215,8 +215,11 @@ pub type DependencyConstraints<P, V> = Map<P, Range<V>>;
 /// Trait that allows the algorithm to retrieve available packages and their dependencies.
 /// An implementor needs to be supplied to the [resolve] function.
 pub trait DependencyProvider<P: Package, V: Version> {
-    /// Decision making is the process of choosing the next package
-    /// and version that will be appended to the partial solution.
+    /// This method chooses the next package and its version that will be tested whether
```

Comment:
A bit awkward said like that. We've already determined that any existing package version within the constraint is valid so "that will be tested whether ..." is more confusing in my opinion. We should just stick to say that we choose any existing package version within the provided constraints.

Reply:
I've restored the wording that was there before the change, just added a link to Pubgrub docs regarding the process.